exec/src/main/java/com/optum/sourcehawk/exec/FlattenConfigCommand.java

```diff
@@ -0,0 +1,177 @@
+import com.optum.sourcehawk.core.constants.SourcehawkConstants;
+import com.optum.sourcehawk.core.repository.LocalRepositoryFileWriter;
+import com.optum.sourcehawk.core.scan.FlattenConfigResult;
+import com.optum.sourcehawk.core.utils.StringUtils;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.Callable;
+
+/**
+ * CLI entry point for executing Sourcehawk flatten command
+ *
+ * @author Christian Oestreich
+ */
+@Slf4j
+@CommandLine.Command(
+        name = FlattenConfigCommand.COMMAND_NAME,
+        aliases = {"flatten", "flat", "merge", "converge", "fc"},
+        description = "Flatten the sourcehawk configuration onto system by recursively reading and merging local and remote configurations will output to console by default",
+        mixinStandardHelpOptions = true
+)
+class FlattenConfigCommand implements Callable<Integer> {
+
+    private static final ObjectWriter WRITER_WITH_DEFAULT_PRETTY_PRINTER = Sourcehawk.YAML_FORMATTER.writerWithDefaultPrettyPrinter();
+
+    static final String COMMAND_NAME = "flatten-config";
+
+    @CommandLine.Option(
+            names = {"-o", "--output"},
+            description = "Optional param to directly output flattened configuration to file system"
+    )
+    private Path output;
+
+    @CommandLine.ArgGroup
+    protected AbstractCommand.ConfigFileExclusiveOptions configFile;
+
+    /**
+     * Bootstrap the command
+     *
+     * @param args the command line args
+     */
+    public static void main(final String... args) {
+        val command = new FlattenConfigCommand();
+        val status = new CommandLine(command)
+                .setCaseInsensitiveEnumValuesAllowed(true)
+                .setTrimQuotes(true)
+                .execute(args);
+        Runtime.getRuntime()
+                .halt(status);
+    }
+
+    /**
+     * flatten the configuration
+     *
+     * @return the exit code
+     */
+    public Integer call() {
+        FlattenConfigResult flattenResult;
+        String configurationFileLocation = getConfigurationFileLocation();
+        if (StringUtils.isBlankOrEmpty(configurationFileLocation)) {
+            return CommandLine.ExitCode.SOFTWARE;
+        }
+        try {
+            flattenResult = ConfigurationReader.readConfiguration(Paths.get("."), configurationFileLocation)
+                    .map(sourcehawkConfiguration -> executeFlatten(configurationFileLocation, sourcehawkConfiguration))
+                    .orElseGet(() -> FlattenConfigResult.error(String.format("Configuration file %s not found or invalid", configurationFileLocation)));
+        } catch (final Exception e) {
+            flattenResult = FlattenConfigResult.error(e.getMessage());
+        }
+
+        outputResults(flattenResult, output);
+
+        if (flattenResult.isError()) {
```

Comment:
Don't we want to check this before output?

Reply:
yes